streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -125,40 +126,21 @@ public boolean isValidTransition(final State newState) {
 
     void postCommit();
 
-    /**
```

Comment:
I'm wondering if we could merge `committableOffsetsAndMetadata` with `prepareCommit` as well, letting the latter to return the map? See my other comment aside.

Reply:
That is certainly possible. Good catch!